torchgeo/trainers/detection.py

```diff
@@ -0,0 +1,249 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Detection tasks."""
+
+from typing import Any, Dict, cast
+
+import pytorch_lightning as pl
+import torch
+import torchmetrics
+import torchvision
+from packaging.version import parse
+from torch import Tensor
+from torch.optim.lr_scheduler import ReduceLROnPlateau
+from torch.utils.data import DataLoader
+from torchvision.models.detection import FasterRCNN
+from torchvision.models.detection.backbone_utils import resnet_fpn_backbone
+from torchvision.models.detection.rpn import AnchorGenerator
+from torchvision.ops import MultiScaleRoIAlign
+
+if parse(torchmetrics.__version__) >= parse("0.8.0"):
+    from torchmetrics.detection.mean_ap import MeanAveragePrecision
+else:
+    from torchmetrics.detection.map import MeanAveragePrecision  # type: ignore
```

Comment:
What is the type ignore here for?

Reply:
The following mypy error gets triggered without it 

```
torchgeo/trainers/detection.py:24: error: Name "MeanAveragePrecision" already defined (possibly by an import)  [no-redef]
```

Reply:
Can we use `# type: ignore[no-redef]`? I try to be as specific as possible with type ignores. Doesn't matter if it causes the line to take up multiple lines.